pkg/envoy/bootstrap/config.go

```diff
@@ -278,6 +279,11 @@ func (b *Builder) Build() (*xds_bootstrap.Bootstrap, error) {
 							},
 						},
 					},
+					UpstreamConnectionOptions: &xds_cluster.UpstreamConnectionOptions{
```

Comment:
so reading the defaults for the other values present here we have:

* 60s of no activity before sending a keep-alive
* 9 failed probes before terminating
* 75s between probes

that gives us 660s, or 11 minutes before we disconnect, which feels too long. I'd recommend we bring this down to 5 probes and 5 seconds, which gives us 80s to terminate which is much better.

Reply:
That sounds reasonable. Did you mean 5 second interval or 5 second before the first probe goes out?

Reply:
5s interval